### PR TITLE
Capability: Add PhoneCall interface v1.2

### PR DESCRIPTION
**Pull request by JeanTracker**
PhoneCallAgent is available to dial, hangup and answer the call.
Please see document as below.

<https://developers-doc.nugu.co.kr/nugu-sdk/capability-interface/phonecall>

Signed-off-by: JeanTracker <hyojoong.kim.jean@gmail.com>

### CODE REVIEW

File: include/capability/phone_call_interface.hh [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+class IPhoneCallListener : public ICapabilityListener {
+public:
+    virtual ~IPhoneCallListener() = default;
+
+    /**
+     * @brief Report changes in the phone call state to the user.
+     * @param[in] state phone call state
+     */
+    virtual void callStateChanged(PhoneCallState state) = 0;
+    /**
```
Comment (webispy) [inline include/capability/phone_call_interface.hh:61]: How about adding a blank line between method declarations?

File: include/capability/phone_call_interface.hh [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+     * @brief The result of send candidates process
+     * @param[in] context_template template information in context
+     * @param[in] payload event's payload
+     */
+    virtual void candidatesListed(const std::string& context_template, const std::string& payload) = 0;
+    /**
+     * @brief The succeed result of send message process
+     * @param[in] payload event's payload
+     */
+    virtual void callArrived(const std::string& payload) = 0;
```
Comment (webispy) [inline include/capability/phone_call_interface.hh:106]: Comment for API is "send message process", but API name is "callArrived".
Is it right?

File: include/capability/phone_call_interface.hh (hunk 1 above)
Comment (JeanTracker) [reply include/capability/phone_call_interface.hh:61]: Ok. I'll fix it.

File: include/capability/phone_call_interface.hh (hunk 2 above)
Comment (JeanTracker) [reply include/capability/phone_call_interface.hh:106]: Thanks. There is some miss comment.

File: include/capability/phone_call_interface.hh [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+     * @brief Report changes in the phone call state to the user.
+     * @param[in] state phone call state
+     */
+    virtual void callStateChanged(PhoneCallState state) = 0;
+
+    /**
+     * @brief Process send candidates
+     * @param[in] payload directive's payload
+     */
+    virtual void processSendCandidataes(const std::string& payload) = 0;
```
Comment (kimhyungrok) [inline include/capability/phone_call_interface.hh:66]: The "Candidataes" is misspelling. I think that "Candidates" is correct word.

Comment (JeanTracker) [reply include/capability/phone_call_interface.hh]: Thanks. I'll fixed it.

**Review by kimhyungrok — APPROVED** [review]

**Review by webispy — APPROVED** [review]

**Pull request merged**